jobbergate-api/jobbergate_api/apps/job_scripts/job_script_files.py

```diff
@@ -0,0 +1,269 @@
+"""
+Provide a convenience class for managing job-script files.
+"""
+
+from pathlib import Path
+from typing import Any, Dict, List, cast
+
+from buzz import Buzz, require_condition
+from file_storehouse import FileManager
+from jinja2 import Template
+from loguru import logger
+from pydantic import BaseModel, root_validator
+
+from jobbergate_api.apps.applications.application_files import ApplicationFiles
+from jobbergate_api.apps.applications.schemas import ApplicationConfig
+from jobbergate_api.config import settings
+from jobbergate_api.s3_manager import IO_TRANSFORMATIONS, file_manager_factory, s3_client
+
+JOBSCRIPTS_WORK_DIR = "job-scripts"
+JOBSCRIPTS_MAIN_FILE_FOLDER = "main-file"
+JOBSCRIPTS_SUPPORTING_FILES_FOLDER = "supporting-files"
+
+
+class JobScriptCreationError(Buzz):
+    """Raise exception when facing any error creating a job-script from application files."""
+
+
+def flatten_param_dict(param_dict: Dict[str, Any]) -> Dict[str, Any]:
+    """
+    Flatten an input dictionary to support the rendering process.
+    """
+    param_dict_flat = {}
+    for (key, value) in param_dict.items():
+        if isinstance(value, dict):
+            for nest_key, nest_value in value.items():
+                param_dict_flat[nest_key] = nest_value
+        else:
+            param_dict_flat[key] = value
+    return param_dict_flat
+
+
+def inject_sbatch_params(job_script_data_as_string: str, sbatch_params: List[str]) -> str:
+    """
+    Inject sbatch params into job script.
+
+    Given the job script as job_script_data_as_string, inject the sbatch params in the correct location.
+    """
+    logger.debug("Preparing to inject sbatch params into job script")
+
+    if not sbatch_params:
+        logger.warning("Sbatch param list is empty")
+        return job_script_data_as_string
+
+    first_sbatch_index = job_script_data_as_string.find("#SBATCH")
+    string_slice = job_script_data_as_string[first_sbatch_index:]
+    line_end = string_slice.find("\n") + first_sbatch_index + 1
+
+    inner_string = ""
+    for parameter in sbatch_params:
+        inner_string += "#SBATCH " + parameter + "\n"
+
+    new_job_script_data_as_string = (
+        job_script_data_as_string[:line_end] + inner_string + job_script_data_as_string[line_end:]
+    )
+
+    logger.debug("Done injecting sbatch params into job script")
+    return new_job_script_data_as_string
+
+
+class JobScriptFiles(BaseModel):
+    """
+    Model containing job-script files.
```

Comment:
Can you add some documentation here of what the folder structure looks like?

Reply:
Adressed in 5f586c3c4c99d4ce57d6bde77a9eca7914cb04ca

Do you have any extra comment on this?